network/port/port.go

```diff
@@ -0,0 +1,145 @@
+package port
+
+import (
+	"fmt"
+	"net"
+)
+
+const maxPort = 65535
+const maxTries = maxPort - 10000
+
+// Checker is used to check for available ports on a host.
+type Checker struct {
+	addr      *net.TCPAddr
+	listeners []*net.TCPListener
+	ports     map[int]bool
+}
+
+// NewChecker returns a Checker that can check ports on the given host.
+func NewChecker(host string) (*Checker, error) {
+	addr, err := net.ResolveTCPAddr("tcp", host+":0")
+	if err != nil {
+		return nil, err
+	}
+
+	return &Checker{
+		addr:  addr,
+		ports: make(map[int]bool),
+	}, nil
+}
+
+// AvailableRange asks the operating system for ports until it is given a
+// contiguous range of ports size long. It returns the first and last of those
+// port numbers, having released all the ports it took, so they are ready for
+// you to use.
+//
+// NB: there is the potential for a race condition here, where once released,
+// another process gets one of the ports before you use it, so start listening
+// on all the returned ports as soon as possible after calling this.
+func (c *Checker) AvailableRange(size int) (int, int, error) {
+	var err error
+
+	defer func() {
+		err = c.release(err)
+	}()
+
+	for i := 0; i < maxTries; i++ {
+		port, erra := c.availablePort()
+		if erra != nil {
+			err = fmt.Errorf("%w; %s", err, erra.Error())
+
+			continue
+		}
+
+		if set, has := c.checkRange(port, size); has {
+			min, max := firstAndLast(set)
+
+			return min, max, err
```

Comment:
Can declare min, max outside the for loop, break the loop here and return min, max, err at the end.

Reply:
done